festim/exports/trap_density_xdmf.py

```diff
@@ -16,13 +16,27 @@ def __init__(self, trap, **kwargs) -> None:
 
         self.trap = trap
 
-    def write(self, t):
+    def write(self, t, dx, log_level):
         """Writes to file
 
         Args:
             t (float): the time
+            dx (fenics.Measure): the measure for dx
         """
         functionspace = self.function.function_space().collapse()
-        density_as_function = f.project(self.trap.density[0], functionspace)
-        self.function = density_as_function
+        u = f.Function(functionspace)
+        v = f.TestFunction(functionspace)
+        F = f.inner(u, v) * dx
+
+        for mat in self.trap.materials:
+            if isinstance(mat, str):
+                F -= f.inner(self.trap.density[0], v) * dx
```

Comment:
This cannot work.

If you do this, then this is going back to the old behaviour.

Imagine a test case where the materials of a trap are [mat1, "mat_id2"] where mat1 is a festim.Material.

Then your formulation is something like

u = trap_density + trap_density in mat1 and trap_density in mat2